_posts/chen_chen/2017-06-22-Using-Jump-Servers-in-SSH.md

```diff
@@ -0,0 +1,123 @@
+
+## Setting up a Public and Private Key-Pair
+
+To generate a key-pair, enter the following in terminal.
+ 
+~~~BASH
+ssh-keygen
+~~~
+
+The tool will create a public key and a password-protected private key and place them in the folder of your choice (usually ~/.ssh/). 
+
+<span style="color:gold">**MAJOR KEY ALERT!!!**</span> Once keys are generated, you need to tell your client about them using ssh-add. This is needed for connecting to servers and agent forwarding.
+
+~~~BASH
+ssh-add ~/.ssh/id_rsa
+~~~
+
+### Granting Access to a Server
+
+Adding access to a server is as easy as adding the client’s public key (by default located in id_rsa.pub file), to the ‘authorized_keys’ on the server.
+
+~~~BASH
+.ssh/authorized_keys
+~~~
+
+## Jump Servers
+
+A jump server’s main purpose is to bridge communication between the local computer and another server. Using SSH to connect to jump servers creates an end to end encrypted segway for information flow between the local computer and a server. Using a jump server, security of a server network could be improved since only the jump server would be exposed directly to the Internet (and to attacks from outside). This limits the exposure of networks, thereby providing a blanket of SSH protection for all the other nodes. However, jump servers can be a double edged sword since an entire network of servers can be exposed if the jump server is compromised. 
+
+<figure>
+    <img src="{{site.urlimg}}chen_chen/JumpServer.png" />
+    <figcaption>Accessing a network of servers using a jump server</figcaption>
+</figure>
+
+## Agent Forwarding
+
+When establishing an SSH connection with default parameters, the private key is only kept on the local computer and the actual communication between the server and computer is using a challenge string. 
+ 
+However, since the private key is only stored on the user computer, there is a problem when trying to establish a connection between a jump server to another remote server due to the fact that the jump server does not contain the private key. The solution to avoid storing keys on the jump server and compromising security is using an agent on each connection. Using agent forwarding, multiple servers can be jumped through   
+ 
+Despite the name including the term “forwarding” the actual mechanism for agent forwarding does not move the key at all. What the agent does is act as a redirect for when the private key is challenged. Since the private key must never leave the local computer, the agent forwards the challenge down each level of the SSH connection until it reaches the client local machine. Once the challenge is encrypted by the private key, the response is then forwarded by the key back to the server that issued the challenge.
```

Comment:
Should the following reads as '... forwarded by the *agent* back to the server ...'

> the response is then forwarded by the key back to the server that issued the challenge